cobalt/base/process/process_metrics_helper_test.cc

```diff
@@ -0,0 +1,223 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/base/process/process_metrics_helper.h"
+
+#include <atomic>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/bind.h"
+#include "base/containers/contains.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/threading/thread.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace base {
+
+namespace {
+
+// static std::atomic<bool> keep_busy{true};
+
+void BusyWork(std::vector<std::string>* vec) {
+  for (int i = 0; i < 100000; ++i) {
+    vec->push_back(NumberToString(i));
+  }
+}
+
+bool DoBusyWork(Thread* thread) {
```

Comment:
So it looks like DoBusyWork returns true once it finds a thread that has measured work, is that right? Could we name it something that indicates that like WorkUntilFoundCPUUsage (or something else you think is good)?